HLTrigger/JetMET/src/HLTMETCleanerUsingJetID.cc

```diff
@@ -97,10 +93,10 @@ void HLTMETCleanerUsingJetID::produce(edm::Event& iEvent, const edm::EventSetup&
         double mex_diff = mex_goodJets - mex_jets;
         double mey_diff = mey_goodJets - mey_jets;
         //double sumet_diff = sumet_goodJets - sumet_jets;  // cannot set sumet...
-        reco::Candidate::LorentzVector p4_diff(mex_diff, mey_diff, 0, sqrt(mex_diff*mex_diff + mey_diff*mey_diff));
+        reco::Candidate::LorentzVector p4_diff(met->front().px()+mex_diff, mey_diff+met->front().py(), 0, sqrt((met->front().px()+mex_diff)*(met->front().px() +mex_diff)+(met->front().py()+mey_diff)*(met->front().py() +mey_diff)));
 
-        reco::CaloMET cleanmet = met->front();
-        cleanmet.setP4(cleanmet.p4() + p4_diff);
+        reco::CaloMET cleanmet ;
+        cleanmet.setP4(p4_diff);
```

Comment:
Well, my comments are no showstoppers, but I thought we could follow up quickly. To clarify a bit what I was aiming at:

1. Renaming: 
Sure, it does not change physics, but why should one retain misleading/wrong variable names?

2. Usability of cleanmet:
If I understand correctly, the "cleanmet" here will contain only the p4-vector information, but none of the CaloMET-specifics that are usually added in https://github.com/cms-sw/cmssw/blob/6b16de370881dd8ef339d34811b3d1e176c02b80/RecoMET/METAlgorithms/src/CaloSpecificAlgo.cc called from https://github.com/cms-sw/cmssw/blob/6b16de370881dd8ef339d34811b3d1e176c02b80/RecoMET/METProducers/src/CaloMETProducer.cc
So the output here uses the CaloMET data format, but only contains a small fraction of the information that is usually associated with this object. However, I guess putting the minimal information here is still better than the old behavior before the PR (retaining the old detail information before JetId-cleaning, which is incorrect when removing jets).